src/AppInstallerCommonCore/Downloader.cpp

```diff
@@ -121,6 +127,138 @@ namespace AppInstaller::Utility
+
+        //Creating destination file
+        std::filesystem::create_directories(dest.parent_path());
+        std::ofstream emptyDestFile(dest);
+        emptyDestFile.close();
+        ApplyMotwIfApplicable(dest);
+        std::ofstream outfile(dest, std::ofstream::binary | std::ofstream::app);
+
+        static const wchar_t* acceptTypes[] = { L"*/*", NULL };
+
+        wil::unique_hinternet session(InternetOpen(
+            L"winget-cli",
+            INTERNET_OPEN_TYPE_PRECONFIG,
+            NULL,
+            NULL,
+            0));
+        THROW_LAST_ERROR_IF_NULL_MSG(session, "InternetOpen() failed.");
+        std::string header = "Content-type: application/json";
+
+        wil::unique_hinternet hConnect(InternetConnect(session.get(), L"pwabuilder-win-chromium-platform.centralus.cloudapp.azure.com", INTERNET_DEFAULT_HTTPS_PORT, NULL, NULL, INTERNET_SERVICE_HTTP,0,0));
+        THROW_LAST_ERROR_IF_NULL_MSG(hConnect.get(), "InternetConnect() failed.");
+        wil::unique_hinternet hRequest(HttpOpenRequest(hConnect.get(), L"POST", L"msix/generate", L"HTTP/1.1", NULL, acceptTypes, INTERNET_FLAG_RELOAD | INTERNET_FLAG_SECURE, 0));
+        THROW_LAST_ERROR_IF_NULL_MSG(hRequest.get(), "HttpOpenRequest() failed.");
+        DWORD headerLength;
+        SIZETToDWord(header.length(), &headerLength);
+        DWORD bodyLength;
+        SIZETToDWord(body.length(), &bodyLength);
+        THROW_LAST_ERROR_IF_MSG(!HttpSendRequestA(hRequest.get(), header.c_str(), headerLength, (void*)body.c_str(), bodyLength), "HttpSendRequestA() failed");
+    
+        // Check http return status
+        DWORD requestStatus = 0;
+        DWORD cbRequestStatus = sizeof(requestStatus);  
+        THROW_LAST_ERROR_IF_MSG(!HttpQueryInfoA(hRequest.get(),
+        HTTP_QUERY_STATUS_CODE | HTTP_QUERY_FLAG_NUMBER,
+        &requestStatus,
+        &cbRequestStatus,
+        nullptr), "Query download request status failed.");
+
+        if (requestStatus != HTTP_STATUS_OK)
+        {
+            AICLI_LOG(Core, Error, << "Download request failed. Returned status: " << requestStatus);
+            THROW_HR_MSG(MAKE_HRESULT(SEVERITY_ERROR, FACILITY_HTTP, requestStatus), "Download request status is not success.");
+        }
+
+        AICLI_LOG(Core, Verbose, << "Download request status success.");
+
+        DWORD dwSize = 4096;
+
+        char lpOutBuffer[4096] = { 0 };
+
+        //Get publisher and package names
+        StringCchPrintfA((LPSTR)lpOutBuffer, dwSize, "pwabuilder-package-publisher");
+        THROW_LAST_ERROR_IF_MSG(!HttpQueryInfoA(hRequest.get(), HTTP_QUERY_CUSTOM,
+            lpOutBuffer, &dwSize, NULL), "Could not get publisher name.");
+  
+        publisher.append(lpOutBuffer);
+
+
+        StringCchPrintfA((LPSTR)lpOutBuffer, dwSize, "pwabuilder-package-name");
+        THROW_LAST_ERROR_IF_MSG(!HttpQueryInfoA(hRequest.get(), HTTP_QUERY_CUSTOM,
+            lpOutBuffer, &dwSize, NULL), "Could not get package name.");

```

Comment:
Probably need to reset dwSize to `ARRAYSIZE(outBuffer)` before making this second call